x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/pagerduty/IncidentEventTests.java

```diff
@@ -69,79 +71,42 @@ public void testPagerDutyXContent() throws IOException {
-                    actualLinks.add(IncidentEventContext.parse(parser));
-                }
-            } else if (IncidentEvent.Fields.IMAGES.match(currentFieldName, parser.getDeprecationHandler())) {
-                // this is an array
-                if (token != XContentParser.Token.START_ARRAY) {
-                    fail("Images was not an array");
-                }
-                while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                    actualImages.add(IncidentEventContext.parse(parser));
-                }
-            } else if (IncidentEvent.Fields.PAYLOAD.match(currentFieldName, parser.getDeprecationHandler())) {
-                // this is a nested object containing a few interesting bits
-                if (token != XContentParser.Token.START_OBJECT) {
-                    fail("payload was not an object");
-                }
-                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-                    if (token == XContentParser.Token.FIELD_NAME) {
-                        currentFieldName = parser.currentName();
-                    } else if (IncidentEvent.Fields.SUMMARY.match(currentFieldName, parser.getDeprecationHandler())) {
-                        actualDescription = parser.text();
-                    } else if (IncidentEvent.Fields.SOURCE.match(currentFieldName, parser.getDeprecationHandler())) {
-                        actualWatchId = parser.text();
-                    } else if (IncidentEvent.Fields.SEVERITY.match(currentFieldName, parser.getDeprecationHandler())) {
-                        actualSeverity = parser.text();
-                    } else if (IncidentEvent.Fields.CUSTOM_DETAILS.match(currentFieldName, parser.getDeprecationHandler())) {
-                        // nested payload object is in here
-                        if (token != XContentParser.Token.START_OBJECT) {
-                            fail("custom_details was not an object");
-                        }
-                        parser.nextToken();
-                        Map<String, Object> mapped = parser.map();
-                        // the first entry is the payload
-                        actualPayload = new Payload.Simple((Map<String, Object>) mapped.get("payload"));
-                    } else {
-                        fail("an unexpected field was encountered inside payload: " + currentFieldName);
-                    }
-                }
-            } else {
-                // this case should not happen
-                fail("an unexpected field was encountered: " + currentFieldName);
+        if (payloadDetails != null) {
+            actualPayload = new Payload.Simple(payloadDetails);
+        }
+
+        List<IncidentEventContext> actualLinks = new ArrayList<>();
+        List<Map<String, String>> linkMap = (List<Map<String, String>>) objectPath.evaluate(IncidentEvent.Fields.LINKS.getPreferredName());
+        if (linkMap != null) {
```

Comment:
`assertThat(linkMap, is(notNullValue))`

Reply:
if the map is not always non-null, then check this implicitely by checking another boolean

Reply:
It can be null if the links are null. They are randomly set to null in the setup code. So this can totally come back as null, which in that case it will still evaluate the images/links empty List's against the setup's empty Lists, ensuring they are the same.